tools/cirdl/cirdl.nim

```diff
@@ -0,0 +1,91 @@
+import std/os
+import pkg/chronicles
+import pkg/chronos
+import pkg/ethers
+import pkg/questionable
+import pkg/questionable/results
+import pkg/zip/zipfiles
```

Comment:
You don't need this and their external dependencies, there is already https://github.com/status-im/nim-zlib which we link against either way already.

Reply:
It would be great to drop this dependency, it's quite annoying... but nim-zlib (and zlib-gzip in general) doesn't do what we need here. The artifact we download is a zip with multiple files. nim-zlib doesn't handle those. So I suppose we can A) rewrite the artifact producing pipeline to a generate .tar.gz file and manually write the code to implement tar handling to reconstruct multiple files using nim-zlib. Or B) use zip/zipfiles.
Got any better ideas?

Reply:
How are this artifacts produced? The more straightforward thing to do is to use tar and gz, option A sounds more reasonable.

Reply:
Well, keep in mind that if we switch to a tar.gz file instead of a zip, nim-zlib doesn't support the tar format. It just does the gzip data streams. We would have to pull in something like https://github.com/guzba/zippy or implement the spec ourselves, or wrap something that implements it.

Reply:
(I also learned that "Nim Tar" is a star wars character.... :D )

Reply:
Switching to any other "bundling format" for the circuit assets is easy. One-line change here: https://github.com/codex-storage/codex-storage-proofs-circuits/blob/master/.github/workflows/generate.yml#L140

So pick what is best suited for your needs.